homeassistant/components/switchbot/config_flow.py

```diff
@@ -151,6 +158,57 @@ async def async_step_password(
             },
         )
 
+    async def async_step_lock_auth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the SwitchBot API auth step."""
+        errors = {}
+        assert self._discovered_adv is not None
+        if user_input is not None:
+            try:
+                key_details = await self.hass.async_add_executor_job(
+                    SwitchbotLock.retrieve_encryption_key,
+                    self._discovered_adv.address,
+                    user_input[CONF_USERNAME],
+                    user_input[CONF_PASSWORD],
+                )
+                return await self.async_step_lock_key(key_details)
```

Comment:
Please only wrap the line that can raise in the try... except block. Use an `else:` block if needed.

Reply:
https://github.com/home-assistant/core/pull/84888